Source/ASTextNode2.mm

```diff
@@ -315,6 +338,98 @@ - (UIAccessibilityTraits)defaultAccessibilityTraits
+  // NO for isAccessibilityElement
+  return NO;
+}
+
+- (NSInteger)accessibilityElementCount
+{
+  if (!ASActivateExperimentalFeature(ASExperimentalTextNode2A11YContainer)) {
+    return [super accessibilityElementCount];
+  }
+
+  return self.accessibilityElements.count;
+}
+
+- (NSArray *)accessibilityElements
+{
+  if (!ASActivateExperimentalFeature(ASExperimentalTextNode2A11YContainer)) {
+    return [super accessibilityElements];
+  }
+
+  NSAttributedString *attributedText = _attributedText;
+  NSInteger attributedTextLength = attributedText.length;
+  if (attributedTextLength == 0) {
+    return @[];
+  }
+
+  NSMutableArray<ASTextNodeAccessiblityElement *> *accessibilityElements = [[NSMutableArray alloc] init];
+
+  // Search the first node that is not layer backed
+  ASDisplayNode *containerNode = self.firstNonLayerNode;
+  ASTextLayout *layout = ASTextNodeCompatibleLayoutWithContainerAndText(_textContainer, attributedText);
+
+  // Create an accessibility element to represent the label's text. It's not necessary to specify
+  // a accessibilityRange here, as the entirety of the text is being represented.
+  ASTextNodeAccessiblityElement *accessibilityElement = [[ASTextNodeAccessiblityElement alloc] initWithAccessibilityContainer:containerNode.view];
+  accessibilityElement.accessibilityLabel = self.accessibilityLabel;
+  accessibilityElement.accessibilityValue = self.accessibilityValue;
+  accessibilityElement.accessibilityTraits = self.accessibilityTraits;
+  if (AS_AVAILABLE_IOS_TVOS(11, 11)) {
+    accessibilityElement.accessibilityAttributedLabel = self.accessibilityAttributedLabel;
+    accessibilityElement.accessibilityAttributedHint = self.accessibilityAttributedHint;
+    accessibilityElement.accessibilityAttributedValue = self.accessibilityAttributedValue;
+  }
+  ASUpdateAccessibilityFrame(accessibilityElement, layout, containerNode, self);
+  [accessibilityElements addObject:accessibilityElement];
+
+  // Collect all links as accessiblity items
+  for (NSString *linkAttributeName in _linkAttributeNames) {
+    [attributedText enumerateAttribute:linkAttributeName inRange:NSMakeRange(0, attributedTextLength) options:NSAttributedStringEnumerationLongestEffectiveRangeNotRequired usingBlock:^(id  _Nullable value, NSRange range, BOOL * _Nonnull stop) {
+      if (value == nil) {
+        return;
+      }
+      ASTextNodeAccessiblityElement *accessibilityElement = [[ASTextNodeAccessiblityElement alloc] initWithAccessibilityContainer:self];
+      accessibilityElement.accessibilityTraits = UIAccessibilityTraitLink;
+      accessibilityElement.accessibilityLabel = [attributedText.string substringWithRange:range];
+      accessibilityElement.accessibilityRange = range;
+      if (AS_AVAILABLE_IOS_TVOS(11, 11)) {
+        accessibilityElement.accessibilityAttributedLabel = [attributedText attributedSubstringFromRange:range];
+      }
+      ASUpdateAccessibilityFrame(accessibilityElement, layout, containerNode, self);
```

Comment:
Nit: should this be a method `-[ASTextNodeAccessibilityElement updateFrameWithLayout:containerNode:textNode:]`? Seems a little out of place as a free function.

Reply:
Done